superset/cli/main.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from typing import Any, Dict
+
+import click
+from colorama import Fore, Style
+from flask.cli import FlaskGroup, with_appcontext
+
+from superset import app, appbuilder, security_manager
+from superset.cli import celery, examples, importexport, test, thumbnails, update
+from superset.cli.lib import feature_flags, normalize_token
+from superset.extensions import db
+
+logger = logging.getLogger(__name__)
+
+
+@click.group(
+    cls=FlaskGroup, context_settings={"token_normalize_func": normalize_token},
+)
+@with_appcontext
+def superset() -> None:
+    """This is a management script for the Superset application."""
+
+    @app.shell_context_processor
+    def make_shell_context() -> Dict[str, Any]:
+        return dict(app=app, db=db)
+
+
+# add sub-commands
+superset.add_command(celery.flower)
```

Comment:
At this point the file is still manageable, but I'm planning to add more commands in the next months (for syncing to/from DBT and other metric stores) and I'm worried it's going to get too big.

I followed the pattern described in the docs (https://click.palletsprojects.com/en/6.x/quickstart/#nesting-commands). Not sure if there's a better way of doing this that doesn't require touching this file when adding a new command. :-/

Reply:
previously we only needed to add has a decorator `superset.command`

Reply:
Right, but I don't think that would work here.

For example, in `cli/celery.py` we could try this:

```python
from superset.cli.main import superset

@superset.command()  # instead of @click.command
def some_command():
    ...
```

But the command wouldn't be registered until we explicitly import the `cli.celery` module. So we'd have to add this to `cli/main.py`:

```python
from superset.cli import celery
```

So we'd have an import that is needed because of an implicit side effect, which IMHO is worse than explicitly having to register everything.

Reply:
@dpgaspar I was able to load them dynamically.